addons/mail/models/mail_thread.py

```diff
@@ -868,6 +868,12 @@ def _routing_check_route(self, message, message_dict, route, raise_exception=Tru
         if alias:
             if thread_id:
                 obj = record_set[0]
+                if 'mail.group' in self.env and isinstance(obj, self.env['mail.group'].__class__):
```

Comment:
For all intents and purposes, this is the same as checking obj._name == 'mail.group' ^^

I feel like that's probably something we'll want to fix for alias mixins at large, but it's difficult to judge the effects it could have..